File: src/main/java/com/pancisin/bookster/models/Category.java
package com.pancisin.bookster.models;

import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonIgnore;

@Entity
@Table(name = "categories")
public class Category {
	
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;
	
	@Column
	private String name;
	
	@JsonIgnore
	@OneToMany(mappedBy = "category")
	private List<Branch> branches;
	
	@Column
	private String code;
	
	public Long getId() {
		return id;
	}

	public String getName() {
		return name;
	}

	public List<Branch> getBranches() {
		return branches;
	}

	public String getCode() {
		return code;
	}
}

File: src/main/kotlin/com/pancisin/bookster/model/Category.kt
package com.pancisin.bookster.model

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.OneToMany
import javax.persistence.Table

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonView
import com.pancisin.bookster.model.Branch
import com.pancisin.bookster.models.views.Summary

@Entity @Table(name = "categories")
data class Category(
  @Id @JsonView(Summary::class) @GeneratedValue(strategy = GenerationType.AUTO)
  val id: Long? = null,

  @Column
  val name: String? = null,

  @JsonIgnore @OneToMany(mappedBy = "category")
  val branches: List<Branch>? = null,

  @Column @JsonView(Summary::class)
  val code: String? = null
)
